src/thunderbots/software/test/ai/navigator/path_planner/theta_star_path_planner.cpp

```diff
@@ -0,0 +1,80 @@
+#include "ai/navigator/path_planner/theta_star_path_planner.h"
+
+#include <gtest/gtest.h>
+
+#include "../shared/constants.h"
+#include "geom/point.h"
+#include "test/test_util/test_util.h"
+
+TEST(TestThetaStarPathPlanner, test_theta_star_path_planner)
+{
+    Field field = ::Test::TestUtil::createSSLDivBField();
+    Ball ball   = Ball({0, 0}, {0, 0}, Timestamp::fromSeconds(5));
+    Point start{0, 0}, dest{-2, -2};
+
+    std::vector<Obstacle> obstacles = std::vector<Obstacle>();
+
+    Timestamp current_time = Timestamp::fromSeconds(123);
+    Robot robot = Robot(3, Point(2.0, 2.0), Vector(0.0, 0.0), Angle::ofRadians(2.2),
+                        AngularVelocity::ofRadians(-0.6), current_time);
+    obstacles.push_back(Obstacle::createRobotObstacleWithScalingParams(robot, .2, .2));
+
+    std::unique_ptr<PathPlanner> planner =
+        std::make_unique<ThetaStarPathPlanner>(field, obstacles);
+
+
+    auto path_points = planner->findPath(start, dest);
+    if (path_points != std::nullopt)
+    {
+        EXPECT_EQ(2, (*path_points).size());
+        printf("\nThe Path is ");
+        for (Point p : *path_points)
+        {
+            {
+                printf("-> (%lf,%lf) ", p.x(), p.y());
+            }
+        }
+        printf("\n");
+    }
+    else
+    {
+        printf("No path\n");
+    }
```

Comment:
It doesn't look like we're actually checking anything in this test?

Reply:
yeah we're not

Reply:
I don't plan to write comprehensive tests here

Reply:
Ok, we should remove tests if they're not actually checking anything. Could we at least add some basic tests that check for things like:
- a path across an empty field is only the start and end point
- a path around an obstacle does not go through the obstacle
- we can successfully find a path if we start inside an obstacle
- we can successfully find a path if the path ends inside an obstacle

Even just some super basic sanity checks like the above will be really helpful to test that we've not totally broken things at comp. If it's a hassle, let me know and I can write them

Reply:
yeah I can improve it a bit

Reply:
This looks much better, looks like we're still not covering the last two cases though? Should be fairly simple tests, but since we've had bugs with that already definitely something important to check for the future.

Also, we don't want to pollute the CI logs, so we should at least comment out the sections that print stuff out and leave a comment indicating what they're printing for any future readers of this code.

Reply:
I have covered the last two cases